File: rubbish_classification_parent/rubbish_classification_web/src/main/java/com/zp/rubbish/utils/MorseZP.java
package com.zp.rubbish.utils;

import java.io.UnsupportedEncodingException;
import java.util.HashMap;
import java.util.Map;
import java.util.regex.Pattern;

/**
 * 基于 Morse思想的加密算法
 *
 * @author zp
 *
 */
public class MorseZP {


	private static Map<Integer, String> enmap;
	private static Map<String, String> demap;

	static {
		enmap = new HashMap<Integer, String>();
		enmap.put(1, ".----");
		enmap.put(2, "..---");
		enmap.put(3, "...--");
		enmap.put(4, "....-");
		enmap.put(5, ".....");
		enmap.put(6, "-....");
		enmap.put(7, "--...");
		enmap.put(8, "---..");
		enmap.put(9, "----.");
		enmap.put(0, "-----");

		demap = new HashMap<String, String>();
		demap.put(".----", "1");
		demap.put("..---", "2");
		demap.put("...--", "3");
		demap.put("....-", "4");
		demap.put(".....", "5");
		demap.put("-....", "6");
		demap.put("--...", "7");
		demap.put("---..", "8");
		demap.put("----.", "9");
		demap.put("-----", "0");

	}

	/**
	 *
	 * @param txt：传入一个utf-8编码的字符串
	 * @return 返回一个加密后的字符串
	 * @throws UnsupportedEncodingException
	 */
	public static String morseZpEncode(String txt) throws UnsupportedEncodingException {
		byte[] bytes = txt.getBytes("utf-8");
		//System.out.println(Arrays.toString(bytes));
		StringBuffer stringBuffer = new StringBuffer("");
		String s = null;
		char c;
		for (int i = 0; i < bytes.length; i++) {
			s = bytes[i] + "";
			for (int j = 0; j < s.length(); j++) {
				c = s.charAt(j);

				if (c == '-') {
					stringBuffer.append("*");
				} else {
					char letter;
					while (true) {
						letter = (char) (122 - (int) (Math.random() * 57));// 随机生成A-z
						if (letter < 91 || letter > 96) {
							break;
						}
					}

					stringBuffer.append(enmap.get(c - '0'));
					stringBuffer.append(letter);
				}
			}
			stringBuffer.append("=");
		}

		return stringBuffer.toString();

	}

	/**
	 *
	 * @param txt 传入一个加密后的文本串
	 * @return 返回一个解密后 编码为utf-8的结果
	 * @throws UnsupportedEncodingException
	 */
	public static String morseZpDecode(String txt) throws UnsupportedEncodingException {
		// String []text=Pattern.compile("[a-zA-Z]").split(txt);
		String[] text = txt.split("=");
		byte[] bs = new byte[text.length];
		int id = 0;// bs 数组坐标
		String[] tex;
		for (int i = 0; i < text.length; i++) {
			if (text[i] == null || text[i].trim().length() == 0)
				continue;
			tex = Pattern.compile("[a-zA-Z]").split(text[i]);
			String te = "";
			if (tex[0].charAt(0) == '*') {
				te = "-";
				tex[0] = tex[0].substring(1);
			}

			for (int j = 0; j < tex.length; j++) {
				if (tex[j] == null || tex[j].trim().length() == 0)
					continue;
				te += demap.get(tex[j]);
			}
			int v = Integer.valueOf(te);
			bs[id++] = (byte) v;

		}
//		for (byte b : bs) {
//			System.out.println(b);
//		}
		return new String(bs, "utf-8");

	}

}

File: rubbish_classification_parent/rubbish_classification_web/src/main/resources/static/js/morsezp.js
		var glj = ["-27", "-71", "-78", "-27", "-98", "-125", "-27", "-100", "-66"];
		var slj = ["-26", "-71", "-65", "-27", "-98", "-125", "-27", "-100", "-66"];
		var khslj = ["-27", "-113", "-81", "-27", "-101", "-98", "-26", "-108", "-74", "-27", "-98", "-125", "-27", "-100", "-66"];
		var yhlj = ["-26", "-100", "-119", "-27", "-82", "-77", "-27", "-98", "-125", "-27", "-100", "-66"];
		var flj=["-23", "-99", "-98", "-27", "-98", "-125","-27", "-100", "-66"];
		var map = {};
		map['.----'] = 1;
		map['..---'] = 2;
		map['...--'] = 3;
		map['....-'] = 4;
		map['.....'] = 5;
		map['-....'] = 6;
		map['--...'] = 7;
		map['---..'] = 8;
		map['----.'] = 9;
		map['-----'] = 0;

		function morseZpDecode(aa) {
			//var bytes = new Array();
			var text = aa.split("=");
			var bs = new Array();
			var id = 0; // bs 数组坐标
			var tex;
			for(var i = 0; i < text.length; i++) {
				if(text[i] == null || text[i].trim().length == 0)
					continue;

				tex = text[i].split(/[a-zA-Z]/);

				var te = "";
				if(tex[0].indexOf("*") != -1) {
					te = "-";
					tex[0] = tex[0].substring(1);
				}

				for(var j = 0; j < tex.length; j++) {
					if(tex[j] == null || tex[j].trim().length == 0)
						continue;
					te += map[tex[j]];
				}

				bs[id++] = te;

			}

			if(myequals(bs,glj)) {
				return 1;
			} else if(myequals(bs,slj)) {
				return 2;
			} else if(myequals(bs,khslj)) {
				return 3;
			} else if(myequals(bs,yhlj)) {
				return 4;
			}else if(myequals(bs,flj)) {
				return 5;
			}else {
				return 6;
			}

		}

	function myequals(arr1,arr2){
		if(arr1.length!==arr2.length) return false;
		for (var i=0;i<arr1.length;i++) {
			if(arr1[i]!==arr2[i])return false;
		}
		return true;
	}
	
